src/test/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinterTest.java

```diff
@@ -0,0 +1,514 @@
+     */
+    private static ArrayList<String> fragmentList = new ArrayList<>(FragmentFingerprinterTest.initialCapacityValue);
+    /**
+     * fragmentFingerprinter
+     */
+    private static FragmentFingerprinter fragmentFingerprinter;
+    /**
+     * Bit fingerprint
+     */
+    private static IBitFingerprint bitFingerprintTest;
+    /**
+     * Count fingerprint
+     */
+    private static ICountFingerprint countFingerprintTest;
+    /**
+     * List only with unique SMILES and without frequencies
+     */
+   private static ArrayList<String> dataForGenerationBitFingerprint;
+    /**
+     * List in which a fragment occurs more than once
+     */
+   private static ArrayList<String> countListOfUniqueSmiles;
+    //</editor-fold>
+    //
+    //<editor-fold desc="Constructor" defaultstate="collapsed">
+    /**
+     * Empty Constructor
+     */
+    public FragmentFingerprinterTest() {
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="BeforeAll method" defaultstate="collapsed">
+    /**
+     * Start generating fingerprinting data.
+     * To create the fingerprints, 2 text files are used here. One of these text files contains the predefined
+     * fragments and the other text file contains the fragments associated with the molecules.
+     * Structure of the text files can be seen in resources folder.
+     *
+     * @BeforeAll ensures that the setUp method is only executed once.
+     *
+     * @throws IOException is thrown if the constructor is unable to open a text file for logging occurred exceptions.
+     */
+    @BeforeAll
+    public static void setUp() throws IOException {
+        BufferedReader tmpFragmentSetReader;
+        BufferedReader tmpMoleculeFragmentsReader;
+        try {
+            tmpFragmentSetReader = new BufferedReader(new FileReader("src/test/resources/de/unijena/cheminf/fragment/fingerprint/FragmentList.txt")); //src/test/resources/de/unijena/cheminf/fragment/fingerprint/FragmentList.txt
+            tmpMoleculeFragmentsReader = new BufferedReader(new FileReader("src/test/resources/de/unijena/cheminf/fragment/fingerprint/MoleculeFragments.txt"));
+        } catch (IOException anException) {
+            throw new IOException("File is not readable!");
+        }
+        /* Read CSV file to obtain fragments used to create the fingerprint. A fragment in the form of
+        unique SMILES is displayed at the first position of each line. Only these unique
+        SMILES from the file are stored in the list.
+         */
+        String tmpLine;
+        String tmpSeparatorComma = ",";
+        try {
+            while ((tmpLine = tmpFragmentSetReader.readLine()) != null) {
+                String[] tmpSmilesOfFragments = tmpLine.split(tmpSeparatorComma);
+                FragmentFingerprinterTest.fragmentList.add(tmpSmilesOfFragments[0]);
+            }
+            FragmentFingerprinterTest.fragmentList.remove(0);
+            tmpFragmentSetReader.close();
```

Comment:
Please include the close statements in a _finally_ block to be sure that it will be closed even if an exception is thrown